src/RateLimit/HasRateLimiting.php

```
@@ -0,0 +1,132 @@
+     */
+    public function hasRateLimitingEnabled(): bool
+    {
+        return $this->rateLimitConfig !== null;
+    }
+
+    /**
+     * Determine if a failed request should be retried.
+     *
+     * Only retries rate limit errors and server errors (5xx).
+     */
+    public function handleRetry(FatalRequestException|RequestException $exception, Request $request): bool
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$request'. (undefined)

(UnusedFormalParameter)